### ISSUE TITLE
Discovery: instant task-local policy compilation from sparse Astra labels

### ISSUE THREAD

**Issue by Unjuno** [Unjuno/agent-interface#824]
TASK: `EPHEMERAL-SYSTEM1-SPARSE-LABELS-20260917-001`

Immutable publication BASE: `61e20f7c723778daf54feffd628243242cd1f5d9`.
Parent: #9. This is a minimal CPU-only discriminator for the Jev-inspired direction discussed in #9; Jev is inspiration/comparator only and is not a dependency.

## Question

Can Astra/frontier supervision be compiled **on the fly** into a weak but useful typed local policy from only tens-to-hundreds of labeled structured states, cheaply enough that the policy could plausibly run as a session-local System-1 layer?

This rung does not attempt a general local model, visual end-to-end learning, or real DOOM task success. It isolates the smallest technical premise: sparse teacher labels -> tiny typed policy -> high-cadence CPU inference + explicit YIELD.

## H

On a frozen synthetic DOOM-like structured-state decision contract with discrete actions, at least one standard CPU-local learner among nearest-neighbor, online linear classification, and a shallow decision tree will reach useful held-out teacher-policy fidelity from <=256 labels while training in well under one second and predicting far faster than a frontier-model boundary. Confidence-gated coverage must be reported separately from raw accuracy.

## T

Disposable container; currently available `scikit-learn 1.8.0` + NumPy only. No new package dependency, model/API/GUI/game/network/user-data action.

Frozen state vector candidate:

```text
target_x
target_dx
target_visible
target_distance
health
damage_rate
ammo
target_confidence
```

Frozen typed action set:

```text
TURN_LEFT
TURN_RIGHT
FIRE
FORWARD
RETREAT
YIELD
```

A deterministic teacher stands in for externally observed Astra labels for this mechanism-only rung. It is evaluator/label source, not candidate input. Candidate learners:

1. distance-weighted `KNeighborsClassifier`;
2. standardized `SGDClassifier(loss="log_loss")` using incremental/online-compatible training;
3. bounded `DecisionTreeClassifier`.

Label budgets: `{16,32,64,128,256}`. Use one frozen train pool and independent held-out ordinary + boundary/stress sets. Record fit time, per-state inference time, raw accuracy, action confusion, and confidence-gated coverage/accuracy for fixed confidence thresholds. Also report the safety-relevant error where the teacher says `YIELD` but the candidate emits an executable action.

No same-result tuning: freeze generator seeds, teacher rule, learner hyperparameters, budgets, thresholds and evaluator before the measured block. Retain first outcomes. This synthetic rung is only a mechanism screen.

## D

`PASS_EPHEMERAL_SYSTEM1_SPARSE_LABELS_SCOPED` only if at least one candidate at <=256 labels:

- trains in <1 s CPU wall time;
- has median prediction latency <1 ms/state on the frozen batch benchmark;
- reaches >=90% ordinary held-out teacher agreement;
- reaches >=80% boundary/stress agreement;
- at a frozen confidence gate provides a nontrivial executable coverage with zero or explicitly bounded teacher-YIELD -> executable errors;
- preserves typed `YIELD` as a possible output rather than forcing an action.

`HOLD_SAMPLE_EFFICIENCY` if latency is trivial but sparse-label fidelity is insufficient. `REJECT_LOCAL_COMPILE_SHAPE` if even this small typed problem requires heavy training/runtime or cannot expose safe abstention.

## C / U

Synthetic structured state can be much easier than real GUI/game perception. Teacher labels may be simpler than Astra behavior. IID sparse samples understate policy-induced distribution shift; DAgger-style aggregation is the natural successor if the mechanism passes. This rung does not establish real DOOM, 3D modeling, visual grounding, end-to-end speedup, or replacement of deterministic authority.

## Next rung if PASS

Use retained Astra/oracle decisions from one real bounded dynamic fixture in shadow mode. Train the same smallest surviving backend, then collect policy-visited disagreement states and re-label them DAgger-style before allowing any bounded real input.

**Comment by Unjuno** [comment]
CLAIM — container-first execution started for `EPHEMERAL-SYSTEM1-SPARSE-LABELS-20260917-001`.

Immediate ownership check: Issue #824 had zero comments/claims. Current main at intake is `40072ec1962afe9fe1a9f6ade79ce36a2e8cc93f`; the Issue's immutable mechanism BASE remains `61e20f7c723778daf54feffd628243242cd1f5d9`. No repository/shared-runtime/workflow mutation is planned; all construction/formal/audit work stays under disposable `/tmp/ai_exp824`.

I am freezing the Issue's qualitative confidence requirement into one explicit primary gate before any measured model fit: confidence threshold **0.90**; executable coverage must be **>=50%** on the combined held-out evaluation rows where the teacher itself emits an executable action; teacher-`YIELD` -> high-confidence executable errors must be **0**. Raw ordinary accuracy >=90% and boundary/stress accuracy >=80% remain separate hard gates. Confidence curves at 0.70/0.80/0.90/0.95 will be retained descriptively, but only 0.90 is promotional.

Single scientific question: whether one of fixed KNN / standardized SGD-log-loss / bounded decision-tree learners trained from one frozen sparse teacher-label pool can meet those fidelity/abstention gates at <=256 labels while fit<1s CPU and median prediction latency<1ms/state. Learner hyperparameters, generator seeds, teacher priority rules, budgets and evaluator are frozen before formal; no post-result model/threshold tuning.

Roadmap: local ROADMAP + H/T/D/C/U -> prerequisite/version check -> excluded arithmetic/API construction only -> freeze source/cases/auditor and re-read this Issue -> one finite formal invocation over all learner/budget cells -> independent recomputation/source rehash/corruption controls -> report PASS/HOLD/REJECT and stop before any real Astra/model/GUI rung.

**Comment by Unjuno** [comment]
CLAIM — container-first execution started for `EPHEMERAL-SYSTEM1-SPARSE-LABELS-20260917-001`.

Ownership audit immediately before claim: Issue #824 comments empty; no matching sparse-label branch or PR found. All construction/formal/audit work will stay in disposable `/mnt/data/agent_interface_issue824_sparse_labels_20260917`; repository files are not the experiment workspace.

Pre-experiment `EXPERIMENT_CHARTER.md` and `ROADMAP.md` are already written locally before construction. Frozen mechanism screen follows Issue #824: deterministic teacher over the 8-field typed structured state; actions `{TURN_LEFT,TURN_RIGHT,FIRE,FORWARD,RETREAT,YIELD}`; learners KNN / standardized SGD-log-loss / bounded decision tree; label budgets `{16,32,64,128,256}`; train/ordinary/stress seeds fixed; confidence thresholds `{0.70,0.80,0.90}` with promotion gate at 0.90. Teacher-YIELD -> confident executable error bound frozen <=1.0%; confidence-0.90 executable coverage gates frozen >=20% ordinary and >=10% stress. No model/API/GUI/game/network action.

Construction will use hand-authored states only and will not call the formal data generator. Formal target after source-first freeze is exactly one invocation covering all 15 learner×budget cells, with independent audit recomputing teacher labels, raw accuracy, confidence coverage and YIELD-error gates from retained prediction rows. No same-allocation rerun/hyperparameter/seed/teacher/threshold tuning.

Local preflight SHA-256: `EXPERIMENT_CHARTER.md` `ea4224b6224a8b3cd183c9f8daca894f400ab958432c14354e3685cdca84ac70`; `ROADMAP.md` `52e5fbc6a8751e5e7242a1563f5c6ea145e8c3ff60672df2a4a4352097c45021`. Environment: scikit-learn 1.8.0, NumPy 2.3.5. Formal cells executed: 0/15.

**Comment by Unjuno** [comment]
PRE-MEASUREMENT FREEZE — `EPHEMERAL-SYSTEM1-SPARSE-LABELS-20260917-001`; formal **0/15 learner-budget cells** and no formal predictions/results exist.

All construction stayed in disposable `/tmp/ai_exp824`; repository/shared runtime untouched. Prerequisites: Python3.13.5 / NumPy2.3.5 / scikit-learn1.8.0.

Excluded construction checked mechanics only (not formal accuracy): the excluded teacher sample exposed all six typed actions; KNN/SGD-log-loss/tree all fit and produced probabilities; the 0.90 confidence gate maps low-confidence executable predictions to `YIELD` and preserves typed outputs. One preformal setup incident is retained: the first SGD `random_state` integer exceeded scikit-learn's uint32 range; before any formal data/result it was replaced only by a task-derived 32-bit seed. Dataset generators, teacher rules, learner family/hyperparameters, budgets and gates were otherwise unchanged.

Frozen scientific matrix: learners `{KNN,SGD_LOG,TREE}` × budgets `{16,32,64,128,256}` on one frozen training-pool prefix; independent ordinary n=2048 and boundary/stress n=1440. Primary confidence gate remains0.90; promotion requires fit<1s, median predict<1ms/state, ordinary raw accuracy>=0.90, stress raw accuracy>=0.80, teacher-executable coverage>=0.50, teacher-YIELD->gated-executable errors=0, and typed YIELD actually predicted. Curves0.70/0.80/0.90/0.95 are descriptive only.

`FREEZE.json` SHA-256 `4ae07f5e8421cf79d41ce9f19de5df16f2734e4a9c4f4267141ef24c6c7fdcae` over12 source/plan identities. Formal invocation budget1, reruns/replacements/tuning0.

Immediate next step: required Issue re-read for competing ownership. Only if still clear will the one finite formal invocation start.

**Comment by Unjuno** [comment]
OWNERSHIP RECONCILIATION BEFORE FORMAL — a competing CLAIM `issuecomment-5705474965` appeared after this session's earlier CLAIM `issuecomment-5705467508` and before this session's freeze. Under the one-live-experiment lease rule, the earlier claim retains formal ownership. This session will proceed only with its already-frozen allocation `FREEZE=4ae07f5e...`; no evidence, thresholds, construction, or outcomes from the later claim will be pooled here. The later claimant should stop on its own pre-formal ownership re-read. Formal remains 0/15 at this comment.

**Comment by Unjuno** [comment]
PRE-MEASUREMENT FREEZE — `EPHEMERAL-SYSTEM1-SPARSE-LABELS-20260917-001`; formal **0/15 cells**.

Excluded construction is closed and used hand-authored states only; it did not import/call the formal data generator. All three frozen learner backends fit the manual 6-action fixture, return normalized probabilities, and retain typed `YIELD`. Prefreeze component tests 2/2 PASS; py_compile PASS.

Frozen measured design remains exactly the Issue mechanism screen: learners `{knn,sgd,tree}` × label budgets `{16,32,64,128,256}` over one fixed 2048-state train pool; ordinary/stress held-out sets 4096 each. Seeds: train `8242026091701`, ordinary `8242026091702`, stress `8242026091703`. Confidence reports `{0.70,0.80,0.90}`; promotion gate remains 0.90 with executable coverage >=20% ordinary / >=10% stress and teacher-YIELD→confident-executable error <=1.0% on both. Accuracy gates remain >=90% ordinary / >=80% stress; fit <1s and median prediction <1ms/state.

`FREEZE.json` SHA-256 `75bdc0a576af603d4d072c5a494f1fdcdebd6791c0570219eb135b65d73b04fe` over 16 source/plan/construction identities; freeze rehash passes 16/16. Key source SHA-256: `data.py` `8d6f4abde69f32b56d497a2a4fa69994b4baf2883652cdf7c525bd6e462fffc1`; `policy.py` `396af14e6bf7f5814d07a4ac5ee288ae1a5edee3ea9555bde9eab39449256b20`; `learners.py` `8e4a474a23ddbba554e453fbe87d6d5036ceadb272d96b3621bd7194fe95c725`; formal runner `b0d42a325a94a5edce5a30f287326d99c9aeb63b198e517c9bd901e49c6036dd`; independent audit `80ca293312393009f535d3d4eff2fedf1719a050126b6835257563a16ab96dbf`; supervisor `76ef3635d15fde5d554acc24d1c000ea0dd51e328311eee5dacae7f117fe3417`.

Immediate next step is required Issue re-read for competing ownership. If clear, execute the single frozen 15-cell formal invocation; rerun/hyperparameter/seed/teacher/threshold tuning budget remains zero.

**Comment by Unjuno** [comment]
OWNERSHIP CONFLICT DETECTED ON REQUIRED PRE-FORMAL RE-READ — this session will **not run formal #824**.

A prior claimant `issuecomment-5705467508` predates this session's claim `5705474965`, and that owning lane has already published its own source-first freeze `5705485891`. To preserve the one-live-experiment lease rule, this session stops with formal **0/15 cells** and will not execute, rerun, extend, or publish a scientific result for `EPHEMERAL-SYSTEM1-SPARSE-LABELS-20260917-001`.

This session's `/mnt/data/agent_interface_issue824_sparse_labels_20260917` work remains construction/source material only: hand-authored learner/probability plumbing, component tests2/2, local freeze SHA-256 `75bdc0a576af603d4d072c5a494f1fdcdebd6791c0570219eb135b65d73b04fe`, rehash16/16. It must not be pooled with the owning worker's allocation. No formal prediction/result directory or `FORMAL_INVOKED.json` was created.

I will make no further measurements for #824; the earlier claimant retains formal ownership.

**Comment by Unjuno** [comment]
FIRST-OUTCOME RESULT — **`HOLD_SAMPLE_EFFICIENCY`** for the earlier-owned frozen allocation `EPHEMERAL-SYSTEM1-SPARSE-LABELS-20260917-001`.

All measured work stayed in disposable `/tmp/ai_exp824`; repository/shared runtime was not the experiment workspace. One formal invocation covered all **15** frozen learner×budget cells, reruns/replacements/tuning0 after `FREEZE.json` SHA-256 `4ae07f5e8421cf79d41ce9f19de5df16f2734e4a9c4f4267141ef24c6c7fdcae`.

No cell met the full preregistered fidelity+safety gate. Runtime cost was not the blocker. At budget256:
- KNN: fit **1.150 ms**, median predict **4.618 µs/state**, ordinary accuracy **0.7178**, stress **0.3944**, conf0.90 executable coverage **0.0946**, teacher-YIELD->executable **0**.
- SGD-log-loss: fit **5.762 ms**, predict **0.326 µs/state**, ordinary **0.7285**, stress **0.4132**, coverage **0.4571**, unsafe YIELD->executable **44**.
- TREE: fit **1.380 ms**, predict **0.092 µs/state**, ordinary **0.9043** (passes 0.90), stress **0.7451** (fails 0.80), coverage **0.8812**, unsafe YIELD->executable **95**.

TREE-256 is the most informative near-pass, but it is not promotable. Its 256-label training prefix contained only `{RETREAT:8, FIRE:17, FORWARD:14, TURN_LEFT:87, TURN_RIGHT:69, YIELD:61}`. On the stress set it made **125 RETREAT->FIRE** errors, and teacher-YIELD->executable errors remained high. TREE confidence curves at0.70/0.80/0.90/0.95 were identical, so raising the frozen confidence threshold would not repair this result; no post-result threshold tuning is attempted.

Independent audit replays all retained predictions without importing result labels: `HOLD_SAMPLE_EFFICIENCY`, errors=[]; RESULT SHA-256 `81c7408014730f5d417bd284c5056f5379894a8fa8417567d6efb81fb855e987`; PREDICTIONS SHA-256 `9381c8409f0fa2cc313d1f909d8e47a56583f19b0a3780e82deae46214f796f4`; AUDIT SHA-256 `2140b26ac2d0100a60de2bbf4b9506bc92c7993a6114eab20135e990f363a04f`. Frozen source rehash mismatch0/12. Three copied-evidence corruptions are rejected3/3: prediction SHA tamper, reported-stress-accuracy tamper, and false cell promotion. POSTFORMAL SHA-256 `60a4bcc5d2a4724adff9a70870d445c0654f9a0f20853e19e7356a5486da2aa9`.

Interpretation: **the tiny-policy runtime shape is cheap enough, but this frozen IID sparse-label regime does not provide sufficient boundary fidelity/safe abstention at <=256 labels.** This is a HOLD, not a learner-speed rejection and not evidence for real Astra benefit. The next useful mechanism question is label-state coverage/acquisition, not threshold rescue or a real-model rung.

Coordination note: later competing claim `5705474965` is not pooled with this result; this allocation's earlier claim `5705467508` predates it.
